packages/aws-cdk-lib/aws-eks/test/alb-controller.test.ts

```diff
@@ -69,3 +69,267 @@ test('throws when a policy is not defined for a custom version', () => {
     version: AlbControllerVersion.of('custom'),
   })).toThrowError("'albControllerOptions.policy' is required when using a custom controller version");
 });
+
+test('can configure custom repository with alb controller version 2.4.2', () => {
```

Comment:
I don't know that all these tests are useful. It's just testing that a certain string shows up in the template, which has already been tested elsewhere. The important thing when adding these new versions is the integ test -- making sure that the versions you are adding can actually be deployed to cloudformation.

Secondly, if we did want all these tests, we can avoid copy/paste fairly easily. This is where `test.each` comes in handy. Something like:

```ts
test.each(['2.4.1', '2.4.2'])('testname', (version) => {
   AlbController.create(stack, {
     cluster,
     version: AlbControllerVersion.of(version), // we know this is equivalent to AlbControllerVersion.V2_4_2
     repository: 'custom',
});
```

But again, I think we can just do away with all these tests and instead add/update an integ test to make sure that the new versions work.